src/openforms/registrations/tasks.py

```diff
@@ -16,6 +17,56 @@
 logger = logging.getLogger(__name__)
 
 
+@app.task
+def pre_registration(submission_id: int) -> None:
+    submission = Submission.objects.get(id=submission_id)
+    if submission.pre_registration_completed:
+        return
+
+    registration_plugin = submission.get_registration_plugin()
+    if not registration_plugin:
+        submission.set_submission_reference()
+        submission.pre_registration_completed = True
+        submission.save()
+        return
```

Comment:
I think this should be wrapped in a `with transaction.atomic()` block? There is a non-zero chance that generated submission references are not unique

Reply:
The `set_submission_reference` function that gets called by `submission.set_submission_reference()` already wraps the saving of the submission reference in a transaction, and it already tries to save submission (so if there is a duplicate reference, it is raised here). If that fails, no other changes have been performed yet. 
So I wonder if wrapping it again is needed?

Reply:
yes, but the subsequent `submission.save` call is not covered and if that would be (implicitly) doing other write queries there's a risk you get partial saves committed to the DB, I think? or if additional steps were added here? IMO it can't hurt to wrap it in an explicit transaction block.